upgrade.go

```diff
@@ -94,17 +94,17 @@ func (b *versionedKVBackend) Upgrade(ctx context.Context, s logical.Storage) err
 			for {
 				time.Sleep(time.Second)
 
+				// If we failed because the context is closed we are
+				// shutting down. Close this go routine and set the upgrade
+				// flag back to 0 for good measure.
+				if ctx.Err() != nil {
```

Comment:
Might as well put this above the time.Sleep, right?

Reply:
I didn't on purpose, since you said that the filtering code will cause it to run and then tear down right away. My thinking here is that by the time the one-second sleep is done we might be torn down and thus exit at this point; otherwise we'll end up hitting the upgradeDone check at least once and likely still print an error to the log.